src/cpu-template-helper/src/fingerprint/compare.rs

```diff
@@ -0,0 +1,99 @@
+// Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use clap::ValueEnum;
+
+use crate::fingerprint::{Fingerprint, FingerprintField};
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    /// Difference detected between source and target.
+    #[error("Difference detected between source and target: {0}")]
+    DiffDetected(String),
+}
+
+pub fn compare(
+    source: Fingerprint,
```

Comment:
what is expected to be the `source` and `target`? Is the `target` the saved version that we compare against and is the `source` the actual value from the host?

Reply:
Discussing with Nikita offline, we decided to use `prev` and `curr` instead. Making the tool too generic might lead to misuse or unintentional use.